File: backend/src/main/java/com/nearchrist/backend/entity/Adoration.java
package com.nearchrist.backend.entity;

import com.fasterxml.jackson.annotation.JsonBackReference;
import jakarta.persistence.*;
import lombok.Data;
import java.time.LocalTime;

@Entity
@Data
@Table(name = "adorations")
public class Adoration {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long adorationId;

    @ManyToOne
    @JoinColumn(name = "state_id")
    @JsonBackReference("state-adorations")
    private State state;

    @ManyToOne
    @JoinColumn(name = "diocese_id")
    @JsonBackReference("diocese-adorations")
    private Diocese diocese;

    @ManyToOne
    @JoinColumn(name = "parish_id")
    @JsonBackReference("parish-adorations")
    private Parish parish;

    @Column(nullable = false)
    private String adorationType;

    private String adorationLocation;

    private String adorationLocationType;

    private String adorationDay;

    private LocalTime adorationStart;

    private LocalTime adorationEnd;
}

File: backend/src/backend/mappers/adoration.clj
(ns backend.mappers.adoration
  (:require [backend.dto.adoration :as dto]
            [backend.entities.adoration :as ent]))

(defn to-dto [entity]
  (dto/map->AdorationDto {:adoration-id (:adoration_id entity)
                          :state-id (:state_id entity)
                          :state-abbreviation (:state_abbreviation entity)
                          :diocese-id (:diocese_id entity)
                          :diocese-name (:diocese_name entity)
                          :parish-id (:parish_id entity)
                          :parish-name (:parish_name entity)
                          :adoration-type (:adoration_type entity)
                          :adoration-location (:adoration_location entity)
                          :adoration-location-type (:adoration_location_type entity)
                          :adoration-day (:adoration_day entity)
                          :adoration-start (:adoration_start entity)
                          :adoration-end (:adoration_end entity)}))

(defn to-dto-list [entities]
  (map to-dto entities))

(defn to-entity [upsert-dto]
  (ent/map->Adoration {:state-id (:state-id upsert-dto)
                       :diocese-id (:diocese-id upsert-dto)
                       :parish-id (:parish-id upsert-dto)
                       :adoration-type (:adoration-type upsert-dto)
                       :adoration-location (:adoration-location upsert-dto)
                       :adoration-location-type (:adoration-location-type upsert-dto)
                       :adoration-day (:adoration-day upsert-dto)
                       :adoration-start (:adoration-start upsert-dto)
                       :adoration-end (:adoration-end upsert-dto)}))